### ISSUE TITLE
exception thrown when using uint8_t built-in conversion on two digit numbers

### ISSUE THREAD

**Issue by alepinio** [jbeder/yaml-cpp#844]
Hi,

I'm getting

```
❯ g++ test.cpp -lyaml-cpp ; ./a.out
yaml-cpp uint8_t built-in conversion:
"0" -> 0
"1" -> 1
"2" -> 2
"3" -> 3
"4" -> 4
"5" -> 5
"6" -> 6
"7" -> 7
"8" -> 8
"9" -> 9
terminate called after throwing an instance of 'YAML::TypedBadConversion<unsigned char>'
  what():  yaml-cpp: error at line 1, column 1: bad conversion
fish: “./a.out” terminated by signal SIGABRT (Abort)
```

with

```c++
❯ cat test.cpp 
#include <iostream>

#include "yaml-cpp/yaml.h"

int main() {
    std::cout << "yaml-cpp uint8_t built-in conversion:" << std::endl;
    for (int i = 0; i < 256; ++i) {
        std::cout << '"';
        std::cout << i;
        std::cout << '"';
        std::cout << " -> ";
        std::cout << YAML::Load(std::to_string(i)).as<uint8_t>();
        std::cout << std::endl;
    }

    return 0;
}
```

THX, Alejandro

**Comment by dota17** [comment]
I think this is normal and ```as<uint8_t>``` is explained as ```as<unsigned char>```. Or ```as<signed char>``` in #848. For this issue, I want to know:
1. whether yaml-cpp should support the  ```as<uint8_t>```/  ```as<int8_t>```? Is this usage scenario necessary?
2. I tried to achieve this feature in https://github.com/dota17/yaml-cpp/pull/5. If needed, we can discuss this implementation and I am willing to submit is PR.

**Comment by dota17** [comment]
@jbeder, do you have any opinion on this？Or just ignore this.

**Comment by alepinio** [comment]
> I think this is normal and `as<uint8_t>` is explained as `as<unsigned char>`. Or `as<signed char>` in #848. For this issue, I want to know:
> 
>     1. whether yaml-cpp should support the  `as<uint8_t>`/  `as<int8_t>`? Is this usage scenario necessary?
> 
>     2. I tried to achieve this feature in [dota17#5](https://github.com/dota17/yaml-cpp/pull/5). If needed, we can discuss this implementation and I am willing to submit is PR.

1. On the one hand, I'm aware that 21st century computers have plentiful memory, but I don't use `uint8_t` for memory saving but rather because it adds a lot of meaning to the code, specially when programming for embedded systems or related software. On the other hand, as other `uintX_t` are supported (are they?), I think this should be supported for simmetry.
2. Implementation is beyond me...I'm just a user of this library.

**Comment by alepinio** [comment]
Currently I'm using `.as<uint32_t>()` and `static_cast<uint8_t>`. I would be pleased if I could use just the built-in conversion and remove the `static_cast`s.

**Comment by jbeder** [comment]
Is the problem that

```
YAML::Load("10").as<uint8_t>();
```

throws an exception?

**Comment by dota17** [comment]
@jbeder  yes. they want to support the as<uint8_t>/as<int8_t> and I try to implement this feature in #864 .

**Comment by jbeder** [comment]
I see. That's fine.

**Comment by dota17** [comment]
Fixed by #864.
